src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -1227,15 +1813,86 @@ public static int GetPositionId(SymbolicRegexMatcher<TSet> matcher, ReadOnlySpan
             }
         }
 
-        /// <summary>
-        /// Interface for optimizations to accelerate search from initial states.
-        /// </summary>
+
         private interface IInitialStateHandler
         {
-            public static abstract bool TryFindNextStartingPosition<TInputReader>(SymbolicRegexMatcher<TSet> matcher, ReadOnlySpan<char> input, ref CurrentState state, ref int pos)
+            public static abstract bool TryFindNextStartingPosition<TInputReader>(SymbolicRegexMatcher<TSet> matcher,
+                ReadOnlySpan<char> input, ref CurrentState state, ref int pos)
                 where TInputReader : struct, IInputReader;
         }
 
+        /// <summary>
+        /// Interface for accelerated states, returns true if position was changed
+        /// </summary>
+        private interface IAcceleratedStateHandler
+        {
+            public static abstract bool TryFindNextStartingPosition<TOptimizedInputReader>(
+                SymbolicRegexMatcher<TSet> matcher, byte[] lookup, ReadOnlySpan<char> input, ref
+                int currentStateId, ref int pos, int initialStateId)
+                where TOptimizedInputReader : struct, IOptimizedInputReader;
+        }
+
+        private readonly struct NoAnchorAcceleratedStateHandler : IAcceleratedStateHandler
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static bool TryFindNextStartingPosition<TOptimizedInputReader>(SymbolicRegexMatcher<TSet> matcher,
+                byte[] lookup,
+                ReadOnlySpan<char> input, ref int currentStateId, ref int pos, int initialStateId)
+                where TOptimizedInputReader : struct, IOptimizedInputReader
+
+            {
+                if (currentStateId != initialStateId)
+                    return false;
```

Comment:
We should at least be consistent intramethod :)
```suggestion
                if (currentStateId != initialStateId)
                {
                    return false;
                }
```